schema/object.go

```diff
@@ -19,11 +23,22 @@ type Object interface {
 // NewObjectSchema creates a new object definition.
 // If you need it tied to a struct, use NewStructMappedObjectSchema instead.
 func NewObjectSchema(id string, properties map[string]*PropertySchema) *ObjectSchema {
+	return newObjectSchema(id, properties, false)
+}
+
+// NewLooseObjectSchema creates a new object definition with loose ID checking.
+// The loose ID checking is useful for generated schemas, where the ID may be insignificant,
+// or could burden workflow development.
+func NewLooseObjectSchema(id string, properties map[string]*PropertySchema) *ObjectSchema {
+	return newObjectSchema(id, properties, true)
+}
+
+func newObjectSchema(id string, properties map[string]*PropertySchema, looseIDMatch bool) *ObjectSchema {
```

Comment:
Sorry, but might it not be best to change the function and variable names that reference the old "loose" terminology, to avoid future confusion?

Reply:
Fixed.